plugins/badges-backend/src/service/router.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+import fetch from 'cross-fetch';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import {
+  errorHandler,
+  PluginEndpointDiscovery,
+} from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { BadgeBuilder, DefaultBadgeBuilder } from '../lib/BadgeBuilder';
+
+export interface RouterOptions {
+  badgeBuilder: BadgeBuilder;
+  logger: Logger;
+  config: Config;
+  discovery: PluginEndpointDiscovery;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const router = Router();
+  const logger = options.logger.child({ plugin: 'badges' });
+  const title = options.config.getString('app.title') || 'Backstage';
+  const catalogUrl = `${options.config.getString('app.baseUrl')}/catalog`;
+  const badgesConfig = options.config.getOptional('badges') ?? {};
+  const badgeBuilder =
+    options.badgeBuilder || new DefaultBadgeBuilder(logger, badgesConfig);
+
+  router.get('/entity/:namespace/:kind/:name/:badgeId', async (req, res) => {
```

Comment:
so the resulting URL is `/entity/default/component/hello-world/34` ? is that right?  should it instead be something like `/badge/:namespace/:kind/:name/` ?

i will admit to NOT being an expert in this area, and the core team may have a way they're suggesting this.  may even just be adding the word badge to the end like `:name/badge/:badgeId` ???

Reply:
I think this works the way you want to as it is. The router gets mounted under `/badges`, and the `badgeId` is whatever you use as key in your app config under `badges`, so it can look like this:
![image](https://user-images.githubusercontent.com/72965/108751456-871c4400-7542-11eb-97be-0908829f5b38.png)

Reply:
What he means is that you have a root path, and then on top of that path you have two things that can match, `badge-specs` and `:badgeId`. If you had a badge ID that was `badge-specs` it would be utterly confusing. So the suggestion is to use `/badge-specs` and `/badge/:badgeId` on top of that root instead, to discern the two